src/main/java/uk/gov/justice/laa/crime/cfecrime/utils/RequestHandler.java

```diff
@@ -1,45 +1,96 @@
+        outcome = getOutcomeFromAgeAndPassportedBenefit(under18, passported);
+
+        CfeCrimeResponse cfeCrimeResponse = new CfeCrimeResponse();
+        if (outcome != null) {
+            cfeCrimeResponse.setOutcome(outcome);
+        }else{
+            statelessApiResponse = cmaService.callCma(cmaRequest);
+
+            setInitialMeansTestOutcome(cfeCrimeRequest, cfeCrimeResponse);
+            if (statelessApiResponse.getInitialMeansAssessment().getResult() == InitAssessmentResult.FULL) {
+                setFullMeansTestOutcome(cfeCrimeRequest, cfeCrimeResponse);
+            }
 
-        CfeCrimeResponse response = new CfeCrimeResponse();
-        if (oc != null) {
-            Result result = new Result();
-            result.setOutcome(oc);
-            response.withResult(result);
         }
-        return response;
+        return cfeCrimeResponse;
     }
 
-    private static Outcome getOutcomeFromAgeAndPassportedBenefit(Boolean clientUnder18, Boolean clientPassportedBenefit){
+    private static Outcome getOutcomeFromAgeAndPassportedBenefit(Boolean clientUnder18, Boolean clientPassportedBenefit) {
         Outcome outcome = null;
-        if (clientUnder18 != null && clientUnder18.booleanValue()){
-            outcome = Result.Outcome.ELIGIBLE;
+        if (clientUnder18 != null && clientUnder18.booleanValue()) {
+            outcome = Outcome.ELIGIBLE_WITH_NO_CONTRIBUTION;
         }
-        if (clientPassportedBenefit !=null && clientPassportedBenefit.booleanValue()){
-            outcome = Result.Outcome.ELIGIBLE;
+        if (clientPassportedBenefit != null && clientPassportedBenefit.booleanValue()) {
+            outcome = Outcome.ELIGIBLE_WITH_NO_CONTRIBUTION;
         }
         return outcome;
     }
+
+    public static void setFullMeansTestOutcome(CfeCrimeRequest cfeCrimeRequest, CfeCrimeResponse cfeCrimeResponse) {
+
+        Outcome fullOutcome = null;
+
+        if (cfeCrimeRequest.getSectionInitialMeansTest() != null){
+            FullAssessmentResult statelessInitialResult = statelessApiResponse.getFullMeansAssessment().getResult();
+            fullOutcome = FullMeansTestOutcomeCalculator.getFullMeansTestOutcome(statelessInitialResult, caseType, magCourtOutcome);
+            cfeCrimeResponse.setOutcome(fullOutcome);
+        }
+        cfeCrimeResponse.setOutcome(fullOutcome);
+
+    }
+
+    public static void setInitialMeansTestOutcome(CfeCrimeRequest cfeCrimeRequest, CfeCrimeResponse cfeCrimeResponse) throws UndefinedOutcomeException {
+
+        Outcome initOutcome = null;
+        if (cfeCrimeRequest.getSectionInitialMeansTest() != null) {
+            caseType = CaseType.valueOf(cfeCrimeRequest.getSectionInitialMeansTest().getCaseType().name());
+            if (cfeCrimeRequest.getSectionInitialMeansTest().getMagistrateCourtOutcome() != null) {
+                magCourtOutcome = MagCourtOutcome.valueOf(cfeCrimeRequest.getSectionInitialMeansTest().getMagistrateCourtOutcome().name());
```

Comment:
```suggestion
                magCourtOutcome = cfeCrimeRequest.getSectionInitialMeansTest().getMagistrateCourtOutcome();
```

Reply:
although again this is creating a side-effect which we would rather like to avoid